src/vcpkg/base/system.process.cpp

```diff
@@ -246,6 +251,100 @@ namespace vcpkg
 #endif
     }
 
+    Optional<ProcessStat> try_parse_process_stat_file(StringView text, StringView origin)
+    {
+        ParserBase p(text, origin);
+
+        p.match_while(ParserBase::is_ascii_digit); // pid
+
+        p.skip_whitespace();
+        p.require_character('(');
+        auto executable = p.match_until([](char32_t c) { return c == ')'; }).to_string();
+        p.next();
+
+        p.skip_whitespace();
+        p.next(); // state
+
+        p.skip_whitespace();
+        auto ppid_str = p.match_while(ParserBase::is_ascii_digit);
+        auto maybe_ppid = Strings::strto<int>(ppid_str);
+        if (!maybe_ppid)
+        {
+            return nullopt;
+        }
+
+        return ProcessStat{maybe_ppid.value_or_exit(VCPKG_LINE_INFO), executable};
+    }
+
+    void get_parent_process_list(std::vector<std::string>& ret)
```

Comment:
Should ret be clear()'d first, or just return the vector given that the caller starts with an empty one?